### ISSUE TITLE
Add feature to invite users to participant team using URL

### ISSUE THREAD

**Issue by deshraj** [Cloud-CV/EvalAI#1346]
## Deliverables

Add a feature to invite new team member to a participant team using a Unique URL. Also, don't forget to write unit tests. :) 

## Steps

1. **user1** enters the email of his teammate (let's say **user2**)
2. Send an email to **user2** (if he/she is registered on EvalAI) with a unique URL
3. **user2** clicks on the URL to accept the invitation to the team.
4. *user1* gets an email saying that "**user2** has accepted your invite to team <blah>" (we can iterate over the email message later) 

### Some important points to note

- [ ] Checks performed before inviting to the team should be similar to current invitation logic

**Comment by isht3** [comment]
@deshraj Can I take this up?

**Comment by RishabhJain2018** [comment]
@isht3 Yes sure ! Go ahead

**Comment by isht3** [comment]
@RishabhJain2018 Cool! On it.

**Comment by isht3** [comment]
@deshraj @RishabhJain2018 Should I implement the front-end for this feature along with the back-end or should I only work on writing back-end now?

**Comment by isht3** [comment]
@deshraj I looked a bit into the issue, from what I understood the aim of the issue to add new users into a team after a participant team is already created, yes? Does that mean if a team is already participating in a challenge, we can add new users to that team so they'll also be able to participate in the challenge?

**Comment by isht3** [comment]
@deshraj I looked a bit into the issue, from what I understood the aim of the issue to add new users into a team after a participant team is already created, yes? Does that mean if a team is already participating in a challenge, we can add new users to that team so they'll also be able to participate in the challenge? OR is it to override the current method of adding directly into a team without the user's permission and send an invite Email so that the user can choose whether to be added onto the team.

**Comment by isht3** [comment]
@deshraj Could you provide some insight? :smile:

**Comment by deshraj** [comment]
You got it right. :)

**Comment by isht3** [comment]
@deshraj @RishabhJain2018  For sending Email invites, can I use the inbuilt `django.core.mail` or should I use something else?

**Comment by deshraj** [comment]
Yep.

**Comment by isht3** [comment]
Hi, @deshraj @RishabhJain2018 https://stackoverflow.com/a/6367458/6439372 This answer suggests using an external SMTP like sendgrid for sending E-mails, I was wondering what we were using for sending E-mail's right now. Should I look into the Sendgrid option?

**Comment by isht3** [comment]
@deshraj Should I do the PR with frontend and backend as one commit?

**Comment by RishabhJain2018** [comment]
@isht3 No, please open different pull requests for frontend and backend.

**Comment by isht3** [comment]
@RishabhJain2018 Okay, cool. But how do I test if it works while hosting?

**Comment by RishabhJain2018** [comment]
Since @isht3 isn't active these days to update the PRs. Hence this issue is still open :)

**Comment by Sanji515** [comment]
@RishabhJain2018 Can I work on this after the completion of my previous issue?

**Comment by yashdusing** [comment]
Even I'd like to work on a medium level difficulty issue but I'm already assigned to a problem. So, it's better that once we complete the assigned problems we work on this

**Comment by RishabhJain2018** [comment]
Hi, @Sanji515 @yashdusing You guys can take this issue only when you've completed the previous ones. Till then it is open for everyone.

**Comment by Sanji515** [comment]
@RishabhJain2018 @deshraj I've completed my previous issue, Can I work on this?

**Comment by RishabhJain2018** [comment]
Sure! you can start with this one now!

**Comment by Happytocode24** [comment]
@RishabhJain2018 can I take this issue.?

**Comment by shashwatdh** [comment]
@deshraj @RishabhJain2018  @Ram81 As the issue is still open, can I work on it ?
